eip-0029.md

```diff
@@ -0,0 +1,67 @@
+# Transaction purpose field
+
+* Author: MrStahlfelge
+* Status: Proposed
+* Created: 18-Feb-2022
+* License: CC0
+* Forking: not needed 
+
+## Motivation 
+
+Most payment systems provide a way to set a "purpose" field, used to display a short message
+from payment issuers to payment receivers.
+
+This EIP describes how one of the optional registers of Ergo boxes can be used to store a purpose
+message.
+This message can be set by wallet applications and dApps and can be shown by Ergo Explorer and
+Wallet applications.
+
+
+## Ergo boxes background
+
+A box can contain up to nine registers. Registers 1 to 3 are reserved and mandatory, registers 4
+to 9 are optional and can be used as input for scripts or are used for token minting (see EIP-4).
+
+Registers must be densely packed. It is not possible to use register 9 without adding register 4
+to 8.
+
+A box cannot be more than 4 kbytes.
+
+## Ergo purpose message standard
+
+Ergo boxes can hold Coll[Byte] that can be a UTF-8 representation of the purpose field.
+However, this would mean that we must designate a certain register to be used as purpose register,
+and because of the restrictions mentioned above this would mean to use R4, blocking it for all
+other use cases. This is not desired.
+
+To overcome this, we define that the purpose is encoded as follows in an arbitrary box:
+
+`Pair([0x50, 0x52, 0x50], Coll[Byte])`
+
+This is a pair of a constant Coll[Byte] used to identify this register as a purpose message
+register, and the actual purpose message as Coll[Byte] (UTF-8 representation).
+This means that, in hex representation, the purpose register always starts with a constant prefix
+of `3c0e0e0350525`.
+
+The first register identified as a purpose message register can be used to read a purpose message
+for this box. If there is no such register on a box, no purpose message is defined.
+
+#### Examples
+
+| Purpose message         | Register hex representation                                        |
+| ----------------------- |:------------------------------------------------------------------:|
+| "Your loan January"     | 3c0e0e0350525011596f7572206c6f616e204a616e75617279                 |
```

Comment:
For sanitization, would it maybe be good to demand the data format encoded be something with known injection protection? Just thinking JSON parsers are known to safely parse JSON, XML parsers as well may, ProtoBuf as well etc.

When a receiving application tries to render the message, if it is a plain string and they don't purposefully sanitize it before rendering, it could present an injection attack vector on that application. If they are forced to use a known safe parser by making the message a specific format, it may create a safety net for applications consuming this field.

Reply:
There should be no injection possible with plain strings as long as they are not handled as something different. Using json serialization to handle a single string is just adding complexity as long as the serialized message is nothing different than a single string.